http/src/main/resources/jupyter/startup.sh

```diff
@@ -54,6 +57,22 @@ if [ "$UPDATE_WELDER" == "true" ] ; then
     docker-compose -f /etc/welder-docker-compose.yaml up -d
 fi
 
+# If a Jupyter start user script was specified, execute it now. It should already be in the docker container
+# via initialization in init-actions.sh (we explicitly do not want to recopy it from GCS on every cluster resume).
+if [ ! -z ${JUPYTER_START_USER_SCRIPT_URI} ] ; then
+  JUPYTER_START_USER_SCRIPT=`basename ${JUPYTER_START_USER_SCRIPT_URI}`
+  log 'Executing Jupyter user start script [$JUPYTER_START_USER_SCRIPT]...'
+  EXIT_CODE=0
+  docker exec --privileged -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_HOME}/${JUPYTER_START_USER_SCRIPT} &> start_output.txt || EXIT_CODE=$?
+  if [ $EXIT_CODE -ne 0 ]; then
+    echo "User start script failed with exit code ${EXIT_CODE}. Output is saved to ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}"
+    retry 3 gsutil -h "x-goog-meta-passed":"false" cp start_output.txt ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}
```

Comment:
I'm wondering what will happen if the start user script has a bug. This call will fail, and the startup script will exit. The cluster will remain in `Starting` until Leo times it out and it moves back to `Stopped`. Is this the behavior we want?
* Should we allow clients to modify the user startup script for `Stopped` clusters so they can iterate without a delete/recreate?
* Should we still startup Jupyter in this case _without_ running the user startup script?

Not sure the answer yet, thinking out loud..

Reply:
Good question. How about on the initial cluster creation? Will it enter the error state if this script fails to execute?

The user script itself is actually not a huge deal in my view. The user already has control over the return code from that script, they can choose whether or not to include `set -e`, for example. So having a non-zero exit code crash things seems good there.

I think it's a general question to apply to this entire script, which could fail transiently for other reasons though. In my opinion the cluster should enter the error state if this happens. Possibly following N stop/start retries by the Leo monitor.